### PR TITLE
[10922] Extend SystemInfo class to provide a method to get environment variables

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

### CODE REVIEW

File: src/cpp/utils/SystemInfo.hpp [hunk 1]
```diff
@@ -71,6 +74,40 @@ class SystemInfo
...
+     * \param [in] env_name the name of the environment variable
+     * \param [out] env_value pointer to the value c-string
+     * @return RETCODE_OK if the environment variable is set.
+     * RETCODE_NO_DATA if the environment variable is unset.
+     * RETCODE_BAD_PARAMETER if the provided parameters are not valid.
+     */
+    static ReturnCode_t get_env(
+            const char* env_name, const char** env_value)
+    {
+        if (env_name == nullptr || env_value == nullptr)
```
Comment (MiguelCompany) [inline src/cpp/utils/SystemInfo.hpp:96]: Should we consider an empty `env_name` as incorrect?

```suggestion
        if (env_name == nullptr || env_value == nullptr || *env_name == '\0')
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
* Fix uncrustify
* Add a unit test

File: src/cpp/utils/SystemInfo.hpp (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply src/cpp/utils/SystemInfo.hpp]: Done in 6c230616a2b92f483864849710847720b666c1c6

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Comment by JLBuenoLopez-eProsima** [comment]
CI failures are unrelated

**Pull request merged**